controllers/managedocs_controller_test.go

```diff
@@ -1114,6 +1187,115 @@ var _ = Describe("ManagedOCS controller", func() {
+			})
+		})
+		When("second notification email addresses is added in add-on parameter", func() {
+			It("should update alertmanager config with the added notification email id", func() {
+				setupAlertmanagerConfigConditions(true, "test-key", true, "test-url", true, []string{"test2@email.com", "test3@email.com"})
+
+				// Wait for alertmanager to get updated with smtp details
+				var resIndex int
+				Eventually(func() string {
+					amconfig := amConfigTemplate.DeepCopy()
+					amconfigKey := utils.GetResourceKey(amconfig)
+					Expect(k8sClient.Get(ctx, amconfigKey, amconfig)).Should(Succeed())
+					for i, reciever := range amconfig.Spec.Receivers {
+						if reciever.Name == "SendGrid" {
+							resIndex = i
+						}
+					}
+					return amconfig.Spec.Receivers[resIndex].EmailConfigs[0].To
+				}, timeout, interval).Should(Equal("test2@email.com, test3@email.com"))
+			})
+		})
+		When("second notification email address in the add-on parameter is removed", func() {
+			It("should update alertmanager config by removing the second email address", func() {
+				secret := addonParamsSecretTemplate.DeepCopy()
+				secretKey := utils.GetResourceKey(secret)
+				Expect(k8sClient.Get(ctx, secretKey, secret)).Should(Succeed())
+
+				// remove notification email from addon param secret
+				delete(secret.Data, "notification-email-1")
+				Expect(k8sClient.Update(ctx, secret)).Should(Succeed())
+
+				// Wait for alertmanager to get updated with smtp details
+				var resIndex int
+				Eventually(func() string {
+					amconfig := amConfigTemplate.DeepCopy()
+					amconfigKey := utils.GetResourceKey(amconfig)
+					Expect(k8sClient.Get(ctx, amconfigKey, amconfig)).Should(Succeed())
+					for i, reciever := range amconfig.Spec.Receivers {
+						if reciever.Name == "SendGrid" {
+							resIndex = i
+						}
+					}
+					return amconfig.Spec.Receivers[resIndex].EmailConfigs[0].To
+				}, timeout, interval).Should(Equal("test2@email.com"))
+			})
+		})
+		When("there is no notification email address in the add-on parameter", func() {
+			It("should update alertmanager config by removing the SMTP email configs", func() {
+				secret := addonParamsSecretTemplate.DeepCopy()
+				secretKey := utils.GetResourceKey(secret)
+				Expect(k8sClient.Get(ctx, secretKey, secret)).Should(Succeed())
+
+				// remove notification email from addon param secret
+				delete(secret.Data, "notification-email-0")
+				Expect(k8sClient.Update(ctx, secret)).Should(Succeed())
+
+				// Wait for alertmanager to remove the email configs
+				var resIndex int
+				Eventually(func() bool {
+					amconfig := amConfigTemplate.DeepCopy()
+					amconfigKey := utils.GetResourceKey(amconfig)
+					Expect(k8sClient.Get(ctx, amconfigKey, amconfig)).Should(Succeed())
+					for i, reciever := range amconfig.Spec.Receivers {
+						if reciever.Name == "SendGrid" {
+							resIndex = i
+						}
+					}
+					return len(amconfig.Spec.Receivers[resIndex].EmailConfigs) == 0
```

Comment:
I was under the impression that the entire receiver will be removed if the last email is removed.